core/services/keeper/registry_synchronizer_process_logs.go

```diff
@@ -53,11 +51,10 @@ func (rs *RegistrySynchronizer) handleSyncRegistryLog(head models.Head, done fun
 	logger.ErrorIf((errors.Wrapf(err, "RegistrySynchronizer: unable to mark log as consumed, jobID: %d", rs.job.ID)))
 }
 
-func (rs *RegistrySynchronizer) handleUpkeepCanceledLogs(head models.Head, done func()) {
+func (rs *RegistrySynchronizer) handleUpkeepCanceledLogs(done func()) {
 	defer done()
-	oldEnough := isOldEnoughConstructor(head, rs.minConfirmations)
 	for {
-		i := rs.mailRoom.mbUpkeepCanceled.RetrieveIf(oldEnough)
```

Comment:
We can probably remove the `RetrieveIf` function from the mailbox struct. I always thought that was a bit hacky anyway.